lib/prometheus_exporter/instrumentation/unicorn.rb

```diff
@@ -26,15 +26,17 @@ def self.start(pid_file:, listener_address:, client:, frequency: 30)
       end
     end
 
-    def initialize(pid_file:, listener_address:)
+    def initialize(pid_file:, listener_address:, metric_labels:)
       @pid_file = pid_file
       @listener_address = listener_address
       @tcp = listener_address =~ /\A.+:\d+\z/
+      @metric_labels = metric_labels
     end
 
     def collect
       metric = {}
       metric[:type] = 'unicorn'
+      metric[:metric_labels] = @metric_labels
```

Comment:
minor, but lets simply not send this over to the other side if it is empty, just allow it to be nil, if nil don't pass it. then in the collector if it is nil you do the old observe

Reply:
I realised we can just use the `custom_labels` set on the client. I've updated the PR accordingly.